gui/job-details.lua

```diff
@@ -0,0 +1,438 @@
+                'Bug Alert',
+                { 'Please set a specific item type first.\n\n',
+                  'Otherwise the material will be matched\n',
+                  'incorrectly due to a limitation in DF code.' },
+                COLOR_YELLOW
+            )
+            return
+        end
+    end
+
+    guimat.MaterialDialog{
+        prompt = 'Please select a new material for input '..idx,
+        none_caption = 'any material',
+        mat_filter = function(mat,parent,mat_type,mat_index)
+            return dfhack.job.isSuitableMaterial(obj.iobj, mat_type, mat_index, obj.iobj.item_type)
+        end,
+        on_select = self:callback('setMaterial', obj)
+    }:show()
+end
+
+function JobDetails:canChangeTrait()
+    local idx, obj = self.list:getSelected()
+    return obj ~= nil and not is_caste_mat(obj.iobj)
+end
+
+function JobDetails:onChangeTrait()
+    local idx, obj = self.list:getSelected()
+    guimat.ItemTraitsDialog{
+        job_item = obj.iobj,
+        prompt = 'Please select traits for input '..idx,
+        none_caption = 'no traits',
+    }:show()
+end
+
+local function ScrJobDetails()
+    return df.global.game.main_interface.job_details
+end
+
+local function show_job_details()
+    local scr = ScrJobDetails()
+    if not scr.open -- dfhack.gui.matchFocusString('dwarfmode/JobDetails')
+    then
+        qerror("This script needs to be run from a job details screen")
+    end
+
+    local job
+    if scr.context == df.job_details_context_type.BUILDING_TASK_LIST then
+        job = scr.jb
+    elseif scr.context == df.job_details_context_type.MANAGER_WORK_ORDER then
+        job = scr.wq
+    end
+    if job == nil then
+        qerror("Unhandled screen context: ".. df.job_details_context_type[scr.context])
+    end
+
+    local dlg = JobDetails{ job = job, context = scr.context }
+    dlg:show()
+end
+
+-- --------------------
+-- DetailsHotkeyOverlay
+--
+
+local focusStrings = 'dwarfmode/JobDetails'
+
+DetailsHotkeyOverlay = defclass(DetailsHotkeyOverlay, overlay.OverlayWidget)
+DetailsHotkeyOverlay.ATTRS{
+    default_pos={x=0,y=0},
+    default_enabled=true,
+    viewscreens=focusStrings,
+    frame={w=1+6+2+(7)+1, h=3},
+}
+
+function DetailsHotkeyOverlay:init()
+    self:addviews{
+        widgets.TextButton{
+            view_id = 'button',
+            frame={t=0, l=0, r=0, h=1},
+            label='details',
+            key='CUSTOM_CTRL_D',
+            on_activate=show_job_details,
+        },
+    }
+end
+
+local function isManagerOrderScreen()
+    local scr = ScrJobDetails()
+    return scr.context == df.job_details_context_type.MANAGER_WORK_ORDER
+end
+
+DetailsHotkeyOverlay_ManagerWorkOrder = defclass(DetailsHotkeyOverlay_ManagerWorkOrder, DetailsHotkeyOverlay)
+DetailsHotkeyOverlay_ManagerWorkOrder.ATTRS{
+    default_pos={x=5, y=5}, -- {x=5, y=5} is right above the job title
+}
+function DetailsHotkeyOverlay_ManagerWorkOrder:onRenderBody(dc)
+    if isManagerOrderScreen() then
```

Comment:
instead of checking here, would it be simpler to attach the overlays to more specific focus strings? The job details context type *is* included, e.g. `dwarfmode/JobDetails/MANAGER_WORK_ORDER`

Reply:
well, now - https://github.com/DFHack/dfhack/pull/3730 - it is :)